backend/contracts/Quiz.sol

```diff
@@ -128,9 +139,67 @@ contract Quiz {
         return (cnt, _allCoupons.length);
     }
 
+    // Generates the permutation vector for all question choices.
+    function getPermutationVector(string memory coupon) private view returns (uint8[] memory) {
+        uint8[] memory pv = new uint8[](_totalChoices);
+
+        uint k=0; // Number of processed choices in total.
+        for (uint i=0; i<_questions.length; i++) {
+            bytes32 seed = keccak256(abi.encode(_key, coupon, i));
+            // Enumerate choices.
+            for (uint8 j=0; j<_questions[i].choices.length; j++) {
+                pv[k+j] = j;
+            }
+            // Permute choices.
+            for (uint j=0; j<_questions[i].choices.length; j++) {
+                uint8 tmp = pv[k+uint8(seed[j%seed.length])%_questions[i].choices.length];
```

Comment:
Maybe we could name `k+uint8(seed[j%seed.length])%_questions[i].choices.length`. I needed a sec to grok this (potentially just a me problem though).